tests/CMakeLists.txt

```diff
@@ -30,6 +30,7 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/../tools")
 
 option(PYBIND11_WERROR "Report all warnings as errors" OFF)
 option(DOWNLOAD_EIGEN "Download EIGEN (requires CMake 3.11+)" OFF)
+option(PYBIND11_CUDA_TESTS "Enable building CUDA tests (requires CMake 3.12+)" OFF)
```

Comment:
Is there a reason our CMake explicitly needs this? Can't it be set during configuation from the outside, by passing `-DCMAKE_CUDA_STANDARD=...`, etc?

Reply:
Yes, because you have to convince CMake that the `.cpp` files are actually CUDA files. Plus a few other things further down, like disabling a test that doesn't run on CUDA.

This only changes the test CMake, though, not the normal/public one.

Reply:
Also, warnings specifications change completely with CUDA.

Reply:
Alright, then.